pkg/hubble/metrics/metrics.go

```diff
@@ -132,44 +119,52 @@ func initEndpointDeletionHandler() {
 	}()
 }
 
-// initMetrics initializes the metrics system
-func initMetrics(address string, metricsTLSConfig *certloader.WatchedServerConfig, enabled api.Map, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) (<-chan error, error) {
-	e, err := initMetricHandlers(enabled)
+// InitMetrics initializes the metrics system
+func InitMetrics(reg *prometheus.Registry, enabled api.Map, grpcMetrics *grpc_prometheus.ServerMetrics) error {
+	e, err := initMetricHandlers(reg, enabled)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	enabledMetrics = e
 
-	registry.MustRegister(grpcMetrics)
-	registry.MustRegister(LostEvents)
-	registry.MustRegister(RequestsTotal)
-	registry.MustRegister(RequestDuration)
+	reg.MustRegister(grpcMetrics)
+	reg.MustRegister(LostEvents)
+	reg.MustRegister(RequestsTotal)
+	reg.MustRegister(RequestDuration)
 
-	errChan := make(chan error, 1)
-
-	initMetricsServer(address, metricsTLSConfig, enableOpenMetrics, errChan)
 	initEndpointDeletionHandler()
 
-	return errChan, nil
+	return nil
 }
 
-// EnableMetrics starts the metrics server with a given list of metrics. This is the
-// function Cilium uses to configure Hubble metrics in embedded mode.
-func EnableMetrics(log logrus.FieldLogger, metricsServer string, metricsTLSConfig *certloader.WatchedServerConfig, m []string, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) error {
-	errChan, err := initMetrics(metricsServer, metricsTLSConfig, api.ParseMetricList(m), grpcMetrics, enableOpenMetrics)
+func EnableMetrics(log logrus.FieldLogger, metricsTLSConfig *certloader.WatchedServerConfig, grpcMetrics *grpc_prometheus.ServerMetrics, enableOpenMetrics bool) {
+	srv := http.Server{
+		Addr:    option.Config.HubbleMetricsServer,
+		Handler: nil,
+	}
+	errChan := make(chan error, 1)
+
+	InitMetricsServerHandler(&srv, Registry, option.Config.EnableHubbleOpenMetrics)
+	err := InitMetrics(Registry, api.ParseMetricList(option.Config.HubbleMetrics), grpcMetrics)
 	if err != nil {
-		return fmt.Errorf("unable to setup metrics: %w", err)
+		log.WithError(err).Error("Unable to setup metrics: %w", err)
+		return
 	}
+
+	go func() {
```

Comment:
I'm still not sure if we should keep the go routine in here at all, and prefer making the caller `launchHubble` handle launching this in a Go routine instead.

Reply:
What's the advantage of moving the Go-routine one level up? 

> In general I think we should favor blocking APIs when possible.

We're still not going to block `launchHubble` on the underlying `listener.Serve()` because of the go-routine, right?

If I move the Go routine to `launchHubble` I'll need to restore the `errChan`, so lmk and I can push the final update, no strong opinion either way, :smile:

Reply:
Generally Go recommends making APIs blocking, and requiring the caller to handle running something async. 
> We're still not going to block launchHubble on the underlying srv.Serve() because of the go-routine, right?

Right.

>  I'll need to restore the errChan, so lmk and I can push the final update, no strong opinion either way, 😄

You wouldn't need the errChan, just the Go routine. You might need to decouple setup of the metrics and starting of the server

It might look like this in `launchHubble`:

```
err = metrics.InitMetrics(metrics.Registry, api.ParseMetricList(option.Config.HubbleMetrics), grpcMetrics)
if err != nil {
  log.WithError(err).Error("Unable to setup metrics: %w", err)
  return
}

go func() {
  if err := metrics.StartMetricsServer(log, metricsTLSConfig, grpcMetrics); err != nil {
    logger.WithError(err).Warn("Hubble metrics server encountered an error")
    return
  }
}()
```

But let's wait for other's feedback.

Reply:
> Generally Go recommends making APIs blocking, and requiring the caller to handle running something async.

I'm inclined to agree here. I actually also think we should pass in a `context.Context` into those Go routine spawning functions so the go routine can be terminated (currently done via `srv.Close` instead, which I guess also works)

Reply:
Made server start function blocking and called from `launchHubble`.